network/ipmasq.go

```diff
@@ -57,3 +57,17 @@ func setupIPMasq(ipn ip.IP4Net) error {
 
 	return nil
 }
+
+func teardownIPMasq(ipn ip.IP4Net) error {
+	ipt, err := iptables.New()
+	if err != nil {
+		return fmt.Errorf("failed to setup IP Masquerade. iptables was not found")
+	}
+
+	err = ipt.DeleteChain("nat", "FLANNEL")
```

Comment:
`teardownIPMasq` should only get called if `setupIPMasq` was called first. If `setupIPMasq` failed, `teardownIPMasq` won't get called. Now, if a user/another program ends up deleting `FLANNEL` chain, it will error (I think correctly) but it'll just get logged and go on.